scripts/vendor-grpc.sh

```diff
@@ -42,41 +43,60 @@ echo "COPYING public gRPC headers"
 for src in "${public_headers[@]}"
 do
 	dest="$DSTROOT/CgRPC/$src"
-	dest_dir=$(dirname $dest)
-	mkdir -pv $dest_dir
-	cp $TMP_DIR/grpc/$src $dest
+	dest_dir=$(dirname "$dest")
+	mkdir -pv "$dest_dir"
+	cp "$TMP_DIR/grpc/$src" "$dest"
 done
 
 echo "COPYING private gRPC headers"
 for src in "${private_headers[@]}"
 do
 	dest="$DSTROOT/CgRPC/$src"
-	dest_dir=$(dirname $dest)
-	mkdir -pv $dest_dir
-	cp $TMP_DIR/grpc/$src $dest
+	dest_dir=$(dirname "$dest")
+	mkdir -pv "$dest_dir"
+	cp "$TMP_DIR/grpc/$src" "$dest"
 done
 
 echo "COPYING gRPC source files"
 for src in "${source_files[@]}"
 do
 	dest="$DSTROOT/CgRPC/$src"
-	dest_dir=$(dirname $dest)
-	mkdir -pv $dest_dir
-	cp $TMP_DIR/grpc/$src $dest
+	dest_dir=$(dirname "$dest")
+	mkdir -pv "$dest_dir"
+	cp "$TMP_DIR/grpc/$src" "$dest"
 done
 
-echo "ADDING additional compiler flags to nanopb/pb.h"
-perl -pi -e 's/\/\* #define PB_FIELD_16BIT 1 \*\//#define PB_FIELD_16BIT 1/' $DSTROOT/CgRPC/third_party/nanopb/pb.h
-
-echo "MOVING nanopb headers to CgRPC/include"
-mv $DSTROOT/CgRPC/third_party/nanopb/*.h $DSTROOT/CgRPC/include/
-
 echo "ADDING additional compiler flags to tsi/ssl_transport_security.cc"
 perl -pi -e 's/#define TSI_OPENSSL_ALPN_SUPPORT 1/#define TSI_OPENSSL_ALPN_SUPPORT 0/' $DSTROOT/CgRPC/src/core/tsi/ssl_transport_security.cc
 
+echo "COPYING CgRPC/src to CgRPC/include"
+cp -rp $DSTROOT/CgRPC/src $DSTROOT/CgRPC/include/src
```

Comment:
Does this need to be copied or would a move suffice? Also, why is this necessary?

Reply:
`src/` contains implementation _and_ headers: we're copying here to maintain the directory structure, we remove all the `.c` and `.cc` files below. There was a symlink from `include/src` to `../src` which I removed because that didn't work with the structure of the `upb-generated` stuff below.